mlkem/reduce.c

```diff
@@ -41,26 +41,41 @@ int16_t montgomery_reduce(int32_t a) {
   // Replace C -> C * q in the above and estimate
   // q / 2^17 < 0.0254.
 
-  uint16_t u;
-  int16_t t;
-  // Compute a*q^{-1} mod 2^16 in unsigned representatives
-  u = (uint16_t)a * QINV;
+  // QINV == -3327 converted to uint16_t == -3327 + 65536 == 62209
+  const uint32_t UQINV = 62209;
```

Comment:
`QINV` is already treated as an unsigned value, so perhaps we should just set `#define QINV 62209`?

Reply:
My general approach is to always use a typed constant where possible rather than a #define so they get type-checked properly. It also shows conscious intent on the part of the programmer.

Reply:
That's too big a change from the ref implementation IMO. But it is true that `QINV` is never used as a signed integer, so its definition as `#define QINV -3327` should be changed to `#define QINV 62209`